### PR TITLE
ebs br: add resilience to single TiKV crash

### PR DESCRIPTION
**Pull request by BornChanger**
### What problem does this PR solve?

This PR provides the resilience to a single TiKV crash due to wal corruption at restore

### What is changed and how does it work?

 Add a new attribute `spec.tolerateSingleTiKVOutage` to restore CR. When it's set to true, restore to the tc can tolerate a single TiKV outage due to wal log corruption issue.

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [x] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/backup/restore/restore_manager.go [hunk 1]
```diff
@@ -399,7 +399,8 @@ func (rm *restoreManager) checkTiFlashAndTiKVReplicasFromBackupMeta(r *v1alpha1.
 		metaInfo.KubernetesMeta.TiDBCluster.Spec.TiKV.Replicas == 0 {
 		klog.Errorf("backup source tc has no tikv nodes")
 		return fmt.Errorf("backup source tc has no tivk nodes")
-	} else if tc.Spec.TiKV.Replicas != metaInfo.KubernetesMeta.TiDBCluster.Spec.TiKV.Replicas {
+	} else if tc.Spec.TiKV.Replicas != metaInfo.KubernetesMeta.TiDBCluster.Spec.TiKV.Replicas ||
+		(r.Spec.TolerateSingleTiKVOutage && metaInfo.KubernetesMeta.TiDBCluster.Spec.TiKV.Replicas-1 == tc.Spec.TiKV.Replicas) {
```
Comment (YuJuncen) [inline pkg/backup/restore/restore_manager.go:403]: I guess we also need to remove this peer or the restore data phase cannot be done.

**Review by WangLe1321 — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

File: pkg/apis/pingcap/v1alpha1/tidbcluster.go [hunk 2]
```diff
@@ -971,8 +971,9 @@ func (tc *TidbCluster) TiKVIsAvailable() bool {
 	return true
 }
 
-func (tc *TidbCluster) AllTiKVsAreAvailable() bool {
-	if len(tc.Status.TiKV.Stores) != int(tc.Spec.TiKV.Replicas) {
+func (tc *TidbCluster) AllTiKVsAreAvailable(tolerateSingleTiKVOutage bool) bool {
+	if (!tolerateSingleTiKVOutage && len(tc.Status.TiKV.Stores) != int(tc.Spec.TiKV.Replicas)) ||
+		(tolerateSingleTiKVOutage && int(tc.Spec.TiKV.Replicas-1) != len(tc.Status.TiKV.Stores)) {
```
Comment (michaelmdeng) [inline pkg/apis/pingcap/v1alpha1/tidbcluster.go:976]: @BornChanger 
would this check return false if we enable `tolerateSingleTiKVOutage` and don't observe a WAL corruption (ie. replicas == stores)? should it be stores >= replicas - 1

Comment (BornChanger) [reply pkg/apis/pingcap/v1alpha1/tidbcluster.go:976]: It's not a common fix but to work-around a single TiKV node crashloop. So, the process is that we observed one TiKV crashes due to the wal corruption issue, THEN turn on tolerateSingleTiKVOutage of the restore CR.

**Pull request merged**